statistics/handle/handle.go

```diff
@@ -845,6 +845,7 @@ func (h *Handle) getStatsReader(history sqlexec.RestrictedSQLExecutor) (*statsRe
 	h.mu.Lock()
 	_, err := h.mu.ctx.(sqlexec.SQLExecutor).Execute(context.TODO(), "begin")
 	if err != nil {
+		h.mu.Unlock()
```

Comment:
it's better to `defer` the unlock operation.

Reply:
Since the statement `return nil, err` has no risk to cause panics, it's ok to don't use `defer` here.

Reply:
is there a chance that `_, err := h.mu.ctx.(sqlexec.SQLExecutor).Execute(context.TODO(), "begin")` paniced during the execution?

Reply:
OK, I update it and add a fail-point test for it.
PTAL @zz-jason